pkg/lint/rules/deprecations.go

```diff
@@ -54,6 +56,11 @@ func validateNoDeprecations(resource *K8sYamlStruct) error {
 		return nil
 	}
 
+	if kubeVersion != nil {
+		k8sVersionMajor = kubeVersion.Major
+		k8sVersionMinor = kubeVersion.Minor
+	}
```

Comment:
This isn't going to work for long running services that consume this as a package. The package level from one run can be changed for a next or concurrent unrelated run. It works for Helm because it's not used for long running services.

This function will need a local variable for the major and minor version. It can use the package level ones as the default which can be overridden by the passed in values.

The package level major/minor are only used in this function so this type of change should be safe.

Reply:
@mattfarina Corrected